From bf854f74d5ad095c012d9cc7389d7a2bc40b035e Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:32:58 -0400
Subject: [alsa] hda: fix format type mismatch in sound/pci/hda/patch_sigmatel.c

Message-id: <140972823573-305-patchset-jkysela@redhat.com>
Patchwork-id: 91001
O-Subject: [RHEL 7.1 PATCH 304/442] ALSA: hda - fix format type mismatch in sound/pci/hda/patch_sigmatel.c
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Signed-off-by: Toralf Foerster <toralf.foerster@gmx.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: e7fc496066ac3cc473f4309d1a1cd32fc2fc08ab)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_sigmatel.c b/sound/pci/hda/patch_sigmatel.c
index 75515b4..7f40a15 100644
--- a/sound/pci/hda/patch_sigmatel.c
+++ b/sound/pci/hda/patch_sigmatel.c
@@ -795,7 +795,7 @@ static int find_mute_led_cfg(struct hda_codec *codec, int default_polarity)
  }
 
  while ((dev = dmi_find_device(DMI_DEV_TYPE_OEM_STRING, NULL, dev))) {
-  if (sscanf(dev->name, "HP_Mute_LED_%d_%x",
+  if (sscanf(dev->name, "HP_Mute_LED_%u_%x",
       &spec->gpio_led_polarity,
       &spec->gpio_led) == 2) {
    unsigned int max_gpio;
@@ -808,7 +808,7 @@ static int find_mute_led_cfg(struct hda_codec *codec, int default_polarity)
     spec->vref_mute_led_nid = spec->gpio_led;
    return 1;
   }
-  if (sscanf(dev->name, "HP_Mute_LED_%d",
+  if (sscanf(dev->name, "HP_Mute_LED_%u",
       &spec->gpio_led_polarity) == 1) {
    set_hp_led_gpio(codec);
    return 1;
-- 
1.7.1